docker/machine.go

```diff
@@ -0,0 +1,246 @@
+		m.container, err = nodes.CreateControlPlaneNode(
+			m.ContainerName(),
+			machineImage(version),
+			clusterLabel(m.cluster),
+			"127.0.0.1",
+			0,
+			nil,
+			nil,
+		)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+	}
+	return nil
+}
+
+// CreateWorker creates a docker container hosting a Kubernetes node.
+func (m *Machine) CreateWorker(version *string) error {
+	// Create if not exists.
+	if m.container == nil {
+		var err error
+		m.Info("Creating worker machine container")
+		m.container, err = nodes.CreateWorkerNode(
+			m.ContainerName(),
+			machineImage(version),
+			clusterLabel(m.cluster),
+			nil,
+			nil,
+		)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+	}
+	return nil
+}
+
+// ExecBootstrap runs bootstrap on a node, this is generally `kubeadm <init|join>`
+func (m *Machine) ExecBootstrap(data string) error {
+	if m.container == nil {
+
+	}
+
+	cloudConfig, err := base64.StdEncoding.DecodeString(data)
+	if err != nil {
+		return errors.Wrap(err, "failed to decode machine's bootstrap data")
+	}
+
+	m.Info("Running machine bootstrap scripts")
+	lines, err := cloudinit.Run(cloudConfig, m.container.Cmder())
+	if err != nil {
+		for _, line := range lines {
+			fmt.Println(line)
+		}
+		return errors.Wrap(err, "failed to join a control plane node with kubeadm")
+	}
+
+	return nil
+}
+
+// SetNodeProviderID patches a node with docker://node-name as the providerID
+func (m *Machine) SetNodeProviderID() error {
+	kubectlNode, err := m.getKubectlNode()
+	if err != nil {
+		return errors.Wrapf(err, "unable to set NodeProviderID. error getting a kubectl node")
+	}
+	if kubectlNode == nil {
+		return errors.New("unable to set NodeProviderID. there are no kubectl node available")
+	}
+
+	m.Info("Setting Kubernetes node providerID")
+	patch := fmt.Sprintf(`{"spec": {"providerID": "%s"}}`, m.ProviderID())
+	cmd := kubectlNode.Command(
+		"kubectl",
+		"--kubeconfig", "/etc/kubernetes/admin.conf",
+		"patch",
+		"node", m.NodeName(),
+		"--patch", patch,
+	)
+	lines, err := exec.CombinedOutputLines(cmd)
+	if err != nil {
+		for _, line := range lines {
+			fmt.Println(line)
+		}
+		return errors.Wrap(err, "failed update providerID")
+	}
+
+	return nil
+}
+
+func (m *Machine) getKubectlNode() (*nodes.Node, error) {
+	// collect info about the existing controlplane nodes
+	kubectlNodes, err := listContainers(
+		withLabel(clusterLabel(m.cluster)),
+		withLabel(roleLabel(constants.ControlPlaneNodeRoleValue)),
+	)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	if len(kubectlNodes) == 0 {
+		return nil, nil
+	}
+
+	return &kubectlNodes[0], nil
+}
+
+// KubeadmReset will run `kubeadm reset` on the machine.
+func (m *Machine) KubeadmReset() error {
+	if m.container != nil {
+		m.Info("Running kubeadm reset on the machine")
+		cmd := m.container.Command("kubeadm", "reset", "--force")
+		lines, err := exec.CombinedOutputLines(cmd)
+		if err != nil {
+			for _, line := range lines {
+				fmt.Println(line)
```

Comment:
This needs to log the line even though I think fmt.Println is a better solution, it probably won't get displayed in the output.